test/fuzz/transformation_load_test.cpp

```diff
@@ -293,6 +293,176 @@ TEST(TransformationLoadTest, BasicTest) {
+  */
+  std::string shader = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %4 "main"
+               OpExecutionMode %4 OriginUpperLeft
+               OpSource ESSL 310
+          %2 = OpTypeVoid
+          %3 = OpTypeFunction %2
+          %6 = OpTypeInt 32 1
+          %7 = OpTypeInt 32 0 ; 0 means unsigned
+          %8 = OpConstant %7 0
+          %9 = OpTypePointer Function %6
+         %10 = OpTypeFunction %2
+         %11 = OpVariable %12 Workgroup
+         %12 = OpTypePointer Workgroup %6
+         %14 = OpConstant %6 0
+         %15 = OpTypePointer Function %6
+         %51 = OpTypePointer Private %6
+         %20 = OpVariable %9 Function
+         %21 = OpConstant %6 2
+         %23 = OpConstant %6 1
+         %25 = OpTypePointer Function %7
+         %50 = OpTypePointer Private %7
+         %34 = OpTypeBool
+         %35 = OpConstantFalse %34
+         %52 = OpVariable %50 Private
+         %53 = OpVariable %51 Private
+          %4 = OpFunction %2 None %3
+          %5 = OpLabel
+         %26 = OpAccessChain %25 %20 %23
+         %30 = OpAccessChain %15 %20 %14
+         %33 = OpAccessChain %15 %20 %14
+               OpSelectionMerge %37 None
+               OpBranchConditional %35 %36 %37
+         %36 = OpLabel
+         %38 = OpAccessChain %15 %20 %14
+         %40 = OpAccessChain %15 %20 %14
+         %43 = OpAccessChain %15 %20 %14
+               OpBranch %37
+         %37 = OpLabel
+               OpReturn
+               OpFunctionEnd
+  )";
+
+  // Making needed instructions
+  const auto env = SPV_ENV_UNIVERSAL_1_4;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Expected (new)transformation after modification (New Args).
+  /*
+
+  TransformationLoad_NEW(
+      uint32_t fresh_id,
+      uint32_t pointer_id,
+      uint32_t execution_scope = 0 (default to avoid crash when working with OpLoad)
+      uint32_t memory_semantic = 0 (default to avoid crash when working with OpLoad)
+      bool opcode_flag *Flag will be false for OpLoad, true for OpAtomicLoad (Is this right?) *
+      const protobufs::InstructionDescriptor& instruction_to_insert_before);
+
+      - opcode_flag: will manage each OpCode
+      e.g.
+      if opcode_flag = false: will call something like OpLoad_processing(....) in IsApplicable or Apply.
+        TransformationLoad_NEW transformation_new (
+         60,
+         11,
+         2,
+         0x100,
+         true,
+         MakeInstructionDescriptor(38, OpAccessChain, 0));
```

Comment:
This looks along the right lines.

I suggset you have:

- uint32_t fresh_id
- uint32_t pointer_id
- bool is_atomic
- uint32_t memory_scope // required to be 0 if is_atomic is false
- uint32_t memory_semantics // required to be 0 if is_atomic is false
- InstructionDescriptor instruction_to_insert_before

I would not use any default parameters.

Reply:
Done.